daggerverse/gcp-secrets/main.go

```diff
@@ -0,0 +1,146 @@
+// A generated module for GcpSecrets functions
+
+package main
+
+import (
+	"context"
+	"dagger/gcp-secrets/internal/dagger"
+	"errors"
+	"strings"
+
+	secretmanager "cloud.google.com/go/secretmanager/apiv1"
+	"cloud.google.com/go/secretmanager/apiv1/secretmanagerpb"
+	"google.golang.org/api/option"
+)
+
+type GcpSecrets struct{}
+
+// Get a single secret from the Google Secret Manager with latest version
+// example: dagger call get-secret --project-id=project-id --secret-name=secret-name --version=latest --credential=path-to-Servcie-account-file
+func (m *GcpSecrets) GetSecretWithVersion(projectID string, secretName string, version string, credential *dagger.File) (string, error) {
```

Comment:
Credential should be type `dagger,Secret`

You don't need to change anything else, Dagger will map it to a file.

Reply:
@rawkode Initially, I tried to use a secret instead of a file, but the issue with that approach is that we encounter parsing errors because it contains special characters. Below is an example of a `SA (Service Account)` token:
```{
  "type": "service_account",
  "project_id": "PROJECT_ID",
  "private_key_id": "KEY_ID",
  "private_key": "-----BEGIN PRIVATE KEY-----\nPRIVATE_KEY\n-----END PRIVATE KEY-----\n",
  "client_email": "SERVICE_ACCOUNT_EMAIL",
  "client_id": "CLIENT_ID",
  "auth_uri": "https://accounts.google.com/o/oauth2/auth",
  "token_uri": "https://oauth2.googleapis.com/token",
  "auth_provider_x509_cert_url": "https://www.googleapis.com/oauth2/v1/certs",
  "client_x509_cert_url": "https://www.googleapis.com/robot/v1/metadata/x509/SERVICE_ACCOUNT_EMAIL"
}
```
One workaround would be to base64 encode these data and pass them as credentials, but the issue is that it would create too much friction for the end-user. Therefore, I decided to use a file instead. Let me know your thoughts on this approach.

Reply:
What code produced that error?

Reply:
@rawkode  I added the new function called `GetSecretViaSecret` that function causing the error the way I run the script is 
1. Exported the credentials to the env variable 
2. Run the below command 
```bash
dagger call get-secret-via-secret --project-id=xxxxxx --secret-name=xxxx --version=latest --credential $CREDS
```
On running that I am getting the below error 
```bash
failed to get value for argument "credential": unsupported secret arg source: "{\n  \"type\""
```

Reply:
Credential needs to use file:path on the command line

Reply:
Example

dagger call github-api --endpoint=https://api.github.com/repos/dagger/dagger/issues --token=file:./github.txt